components/turret.py

```diff
@@ -49,6 +49,7 @@ class Turret:
     def __init__(self):
         # Note that we don't know where the turret actually is until we've
         # seen an index.
+        self.current_state = self.IDLE
```

Comment:
```suggestion
```

Reply:
No. This is necessary here, because is_ready uses it, and is called, presumably by the LED code, before any other initialisation has had a chance to run. Without it, the code crashes with an undefined-attribute error.

Reply:
This is a duplicate line.

Reply:
Ah. It was added by multiple people in slightly different places. Sigh. Sorry.